Tests/Splines/b_spline_test.cpp

```diff
@@ -173,15 +173,17 @@ TEST_F(BSplineSuite, RemoveKnot) {
   BSpline_ b_spline_remove, b_spline, b_spline_unsuccessful;
   ASSERT_NO_THROW(b_spline_remove = BSpline_(parameter_space_remove, vector_space_remove));
   ASSERT_NO_THROW(b_spline = BSpline_(parameter_space, vector_space));
-  ASSERT_NO_THROW(b_spline_unsuccessful = BSpline_(parameter_space_, vector_space_unsuccessful));
+  ASSERT_NO_THROW(b_spline_unsuccessful = BSpline_(parameter_space_unsuccessful, vector_space_unsuccessful));
   EXPECT_EQ(b_spline_remove.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_), kMultiplicity);
   EXPECT_EQ(b_spline_remove, b_spline_insert_remove_);
   EXPECT_EQ(b_spline.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_, kMultiplicity2_), kMultiplicity2_);
   EXPECT_EQ(b_spline, b_spline_);
   EXPECT_EQ(b_spline_remove.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_), kMultiplicity);
   EXPECT_EQ(b_spline_remove, b_spline_);
-  EXPECT_EQ(b_spline_unsuccessful.RemoveKnot(kDimension0_, kKnot0_5_, kEpsilon_), Multiplicity{});
-  EXPECT_EQ(b_spline_unsuccessful, BSpline_(parameter_space_unsuccessful, vector_space_));
+  EXPECT_EQ(b_spline_unsuccessful.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_), Multiplicity{});
+  EXPECT_EQ(b_spline_unsuccessful, b_spline_insert_remove_);
+  EXPECT_EQ(b_spline_.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_), Multiplicity{});
+  EXPECT_EQ(b_spline_, BSpline_(parameter_space_, vector_space_));
```

Comment:
Which knot do we want to remove then? The knot vector for `b_spline_remove` is {[0 0 0 0.5 1 1 1],[0 0 0 0.5 0.5 1 1 1]}.

Reply:
You can choose whatever you like, e.g., `0.25` — just not `0.0`, `0.5`, or `1.0` 😄

Reply:
> 

So in essence we want an additional test on a b-spline level, correct? If we try to remove, say, 0.25 from `b_spline_remove` it will not be equal to `b_spline_`.

Reply:
Let's recap the control flow: calling `BSpline::RemoveKnot` will call `ParameterSpace::RemoveKnot` which in turn will call `KnotVector::Remove`. Thus, we need to tests for removing a non-existent knot for each of these classes. You already adapted the one for `KnotVector` and added the one for `BSpline` (see the code snippet discussed in this conversation). In addition, I asked you to also add such a test for `ParameterSpace` in the general comment of my review.

Coming back to the test for `BSpline`, `b_spline_remove` equals `b_spline` in [line 182](https://github.com/SplineLib/SplineLib/blob/7c936f9c7888d491d9d952278dce007e0be16896/Tests/Splines/b_spline_test.cpp#L182). In this respect, my comment
> You can choose whatever you like, e.g., 0.25 — just not 0.0, 0.5, or 1.0 😄

was wrong, because we could even try to remove `0.5` for dimension 1 at that point. But after attempting to remove a knot that is not contained, `b_spline_remove` should still be equal to `b_spline_`, shouldn't it? So let's try `EXPECT_EQ(b_spline_remove.RemoveKnot(kDimension1_, kKnot0_5_, kEpsilon_), Multiplicity0)` and `EXPECT_EQ(b_spline_remove, b_spline_)` 🤔